### PR TITLE
add normalization guideline in security best practice

### PR DESCRIPTION
**Pull request by yangminzhu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:55]:
```suggestion
not be rejected because it does not match the path defined in the authorization policy due to the extra forward slash `/` in the path.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:59]:
```suggestion
`//` equivalent to a single forward slash `/`.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:55]:
```suggestion
not be rejected because it does not match the path defined in the authorization policy due to the extra forward slash `/` in the path.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:76]:
```suggestion
  to understand which normalization options you may want to use.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:88]:
```suggestion
Before diving into the details of configuring normalization, you should first make sure whether any of the normalizations are needed.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:90]:
```suggestion
You do not need the normalization if you do not use authorization policies or if your authorization policies do not
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:47]:
```suggestion
The enforcement point of authorization policies is the Envoy proxy which is different from the actual resource access point that is
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:81]: Did this section get renamed "Mitigation for unsupported normalization" in [PR #9917](https://github.com/istio/istio.io/pull/9917)?  Also, do you want to add a bullet item for "Safer Authorization Policy Pattern" from that commit?

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:93]:
```suggestion
You may not need the normalization if all your authorization policies follow the safer authorization pattern which, in the worst
```
If you didn't add the "Safer Authorization Policy Pattern" bullet point earlier, I would add a link here to that section.

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:94]:
```suggestion
case, is unexpected rejection instead of policy bypass.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:101]:
```suggestion
This is often the case due to the fact that complicated multi-layered systems make it practically impossible to figure
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (justinpettit) [inline content/en/docs/ops/best-practices/security/index.md:102]:
```suggestion
out which normalization are actually happening to the request beyond the enforcement point.
```

**Review by justinpettit — APPROVED** [review]
Mostly minor grammar nits.  There are a few section headings that I think may be different if you get PR 9917 merged first.  If you don't, you'll probably want a follow-on PR that fixes up the references.

File: content/en/docs/ops/best-practices/security/index.md
Comment (yangminzhu) [reply content/en/docs/ops/best-practices/security/index.md:81]: I was actually just using it as a placeholder. Once the other PR is merged, I will update here and a few other places to refer to the new items.

Comment (yangminzhu) [reply content/en/docs/ops/best-practices/security/index.md]: updated.

**Comment by yangminzhu** [comment]
@istio/wg-docs-maintainers hi, can someone help review the PR? thanks

File: content/en/docs/ops/best-practices/security/index.md
Comment (howardjohn) [inline content/en/docs/ops/best-practices/security/index.md:141]: what is the "safer authorization pattern"? We should link it from here

**Review by howardjohn — COMMENTED** [review]
technical content LGTM

File: content/en/docs/ops/best-practices/security/index.md
Comment (yangminzhu) [reply content/en/docs/ops/best-practices/security/index.md]: added link

**Comment by yangminzhu** [comment]
@ericvn @frankbu @craigbox could any of you help take a look? this has been out for quite some time, thank you.

**Comment by craigbox** [comment]
I have had a quick look at this doc, and it needs more copy editing than the quick review approval I would usually give. We don't have any TWs actively engaged, so I'm not sure if Eric or Frank want to take a pass?

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:96]:
```suggestion
The enforcement point for authorization policies is the Envoy proxy instead of the usual resource access point in the backend application. A policy mismatch happens when the Envoy proxy and the backend application interpret the request
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:99]:
```suggestion
A mismatch can lead to either unexpected rejection or a policy bypass. The latter is usually a security incident that needs to be
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:102]:
```suggestion
For example, consider an authorization policy to reject requests with path `/data/secret`. A request with path `/data//secret` will
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:109]:
```suggestion
In this example, the policy enforcement point (Envoy proxy) had a different understanding of the path than the resource access
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:114]:
```suggestion
* Lack of a clear standard for the normalization.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:118]:
```suggestion
* Applications can even have arbitrary normalizations for their own use cases.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:123]:
```suggestion
* Refer to [Guideline on configuring the path normalization option](/docs/ops/best-practices/security/#guideline-on-configuring-the-path-normalization-option)
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:126]:
```suggestion
* Refer to [Customize your system on path normalization](/docs/ops/best-practices/security/#customize-your-system-on-path-normalization) to
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:129]:
```suggestion
* Refer to [Mitigation for unsupported normalization](/docs/ops/best-practices/security/#mitigation-for-unsupported-normalization) for
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:136]:
```suggestion
Before diving into the details of configuring normalization, you should first make sure that normalizations are needed.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:138]:
```suggestion
You do not need normalization if you don't use authorization policies or if your authorization policies don't
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:141]:
```suggestion
You may not need normalization if all your authorization policies follow the [safer authorization pattern](/docs/ops/best-practices/security/#safer-authorization-policy-patterns)
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:142]:
```suggestion
which, in the worst case, results in unexpected rejection instead of policy bypass.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:150]:
```suggestion
out which normalizations are actually happening to the request beyond the enforcement point.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:152]:
```suggestion
You could use a less strict normalization option if it already satisfies your requirements and you are sure of its implications.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:156]:
```suggestion
For either option, make sure you write both positive and negative tests specifically for your requirements to verify the
normalization is working as expected. The tests are useful in catching potential bypass issues caused by a misunderstanding
or incomplete knowledge of the normalization happening to your request.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:158]:
```suggestion
Refer to [Customize your system on path normalization](/docs/ops/best-practices/security/#customize-your-system-on-path-normalization)
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:159]:
```suggestion
for more details on configuring the normalization option.
```

File: content/en/docs/ops/best-practices/security/index.md
Comment (frankbu) [inline content/en/docs/ops/best-practices/security/index.md:165]:
```suggestion
If you need a specific normalization option that is not supported by Istio yet, please follow
[Mitigation for unsupported normalization](/docs/ops/best-practices/security/#mitigation-for-unsupported-normalization)
for customized normalization support or create a feature request for the Istio community.
```

**Review by frankbu — APPROVED** [review]

**Pull request merged**
